src/org/sosy_lab/java_smt/basicimpl/IndependentInterpolatingProverEnvironment.java

```diff
@@ -127,16 +127,34 @@ public BooleanFormula getInterpolant(Collection<TFormulaInfo> pFormulasOfA)
     BooleanFormula conjugatedA = bmgr.and(formulasOfA);
     BooleanFormula conjugatedB = bmgr.and(formulasOfB);
 
-    // handle empty interpolation groups and trivial interpolants
+    ProverEnvironment prover = getDistinctProver();
+    prover.push(conjugatedA);
+    if (!prover.isUnsat()) {
+      conjugatedA = bmgr.makeTrue();
+    } else {
+      conjugatedA = bmgr.makeFalse();
+    }
+    prover.pop();
+
+    prover.push(conjugatedB);
+    if (!prover.isUnsat()) {
+      conjugatedB = bmgr.makeTrue();
+    } else {
+      conjugatedB = bmgr.makeFalse();
+    }
+    prover.pop();
+
+    // handle empty interpolation groups
     if (formulasOfA.isEmpty()) {
       conjugatedA =  bmgr.makeTrue();
     }
     if (formulasOfB.isEmpty()) {
       conjugatedB = bmgr.makeTrue();
     }
 
+    // handle trivial interpolants
     if ((bmgr.isTrue(conjugatedA) && bmgr.isTrue(conjugatedB))) {
-      return null;
+      throw new IllegalArgumentException("There exists no interpolant.");
```

Comment:
While this is correct, we only allow the users to call `getInterpolant()`  if the last SAT check was UNSAT and the provers stack is unchanged. So this is not needed.

Reply:
I just noticed that you change `conjugatedA` and `conjugatedB` above, making this check wrong based on the wrong assignments above.